src/components/AddressInput.vue

```diff
@@ -0,0 +1,394 @@
+
+        char = char.toUpperCase();
+
+        // Handle I, O, W, Z which are the only characters missing in Nimiq's Base 32 address alphabet
+        switch (char) {
+            case 'I': char = '1'; break;
+            case 'O': char = '0'; break;
+            case 'Z': char = '2'; break;
+            case 'W': return; // reject character
+        }
+
+        const isNumeric = char >= '0' && char <= '9';
+        const isAlphaNumeric = isNumeric || (char >= 'A' && char <= 'Z');
+        if (!isAlphaNumeric
+            || (value === '' && char !== 'N' && !isNumeric) // Must start with N or the checksum (then NQ gets added)
+            || (value === 'N' && char !== 'Q') // Only Q allowed after starting N
+            || (value.length >= 2 && value.length < 4 && !isNumeric)) { // checksum at index 2 and 3 must be digits
+            return; // reject
+        }
+
+        return char;
+    }
+
+    // definiton of the format method for input-format (https://github.com/catamphetamine/input-format#usage)
+    private static _format(value: string) {
+        if (value !== '' && value !== 'N') {
+            // If user typed a valid character and not typed N to start NQ, enforce NQ and form blocks
+            value = AddressInput._stripWhitespace(value)
+                .replace(/^N?Q?/, 'NQ') // enforce NQ at the beginning
+                .replace(/.{4}/g, (match, offset) => `${match}${(offset + 4) % 12 ? ' ' : '\n'}`) // form blocks
+                .substring(0, AddressInput.ADDRESS_MAX_LENGTH); // discarding the new line after last block
+
+            if (value.endsWith(' ')) {
+                // The word spacing set via css is only applied to spaces that are actually between words which is not
+                // the case for an ending space and the caret after an ending space therefore gets rendered at the wrong
+                // position. To avoid that we add a zero-width space as an artificial word. We do not add that to the
+                // template returned to input-format though to avoid it being interpreted as a typed character which
+                // would place the caret after the zero width space.
+                value += '\u200B';
+            }
+        }
+        return {
+            text: value,
+            template: 'xxxx xxxx xxxx\nxxxx xxxx xxxx\nxxxx xxxx xxxx', // used by input-format to position caret
+        };
+    }
+
+    private static _stripWhitespace(value: string) {
+        return value.replace(/\s|\u200B/g, ''); // normal whitespace, tabs, newlines or zero-width whitespace
+    }
+
+    // value that can be bound to via v-model
+    @Prop({
+        type: String,
+        default: '',
+    })
+    public value!: string;
+
+    public $refs: { textarea: HTMLTextAreaElement };
+
+    private currentValue: string = '';
+    private selectionStartBlock: number = -1;
+    private selectionEndBlock: number = -1;
```

Comment:
These are private properties, so shouldn't they also start with an underscore? Just saying :wink:

Reply:
I agree, they should ;)
Unfortunately the vue class component or vue itself doesn't allow for reactive properties starting with an underscore.